source/PlayerInfo.cpp

```diff
@@ -709,10 +738,25 @@ const CoreStartData &PlayerInfo::StartData() const noexcept
 
 
 
+void PlayerInfo::SetSystemEntry(const SystemEntry entryType)
+{
+	entry = entryType;
+}
+
+
+
+SystemEntry PlayerInfo::GetSystemEntry() const
+{
+	return entry;
+}
+
+
+
 // Set the player's current start system, and mark that system as visited.
 void PlayerInfo::SetSystem(const System &system)
 {
 	this->system = &system;
+	this->previousSystem = &system;
```

Comment:
This is just setting previous system and system to the same thing.
```suggestion
	this->previousSystem = this->system;
	this->system = &system;
```

Reply:
~~sorry I changed this too quickly~~